### ISSUE TITLE
musl build failure in threading_pthread.c

### ISSUE THREAD

**Issue by orbea** [DeaDBeeF-Player/deadbeef#2858]
### Steps to reproduce the problem

Build deadbeef with musl.

### What's going on? Describe the problem in as much detail as possible.
```
  CC       deadbeef-threading_pthread.o
threading_pthread.c:100:12: error: incompatible pointer to integer conversion returning 'pthread_t' (aka 'struct __pthread *') from a function with result type 'intptr_t' (aka 'long') [-Wint-conversion]
    return tid;
           ^~~
1 error generated.
make[2]: *** [Makefile:952: deadbeef-threading_pthread.o] Error 1
make[2]: Leaving directory '/home/orbea/src/git/installed/deadbeef'
make[1]: *** [Makefile:1060: all-recursive] Error 1
make[1]: Leaving directory '/home/orbea/src/git/installed/deadbeef'
make: *** [Makefile:635: all] Error 2
```

Full build log: [deadbeef-thread.log](https://github.com/DeaDBeeF-Player/deadbeef/files/9613462/deadbeef-thread.log)

### Information about the software:

Deadbeef version: https://github.com/DeaDBeeF-Player/deadbeef/commit/f0b300b62cd5ac76d6ad9b7ff677f9f3325dd6a8
OS: `Gentoo`
musl: `1.2.3`
clang: `15.0.0`

**Comment by Oleksiy-Yakovenko** [comment]
relates to #2856

**Comment by Oleksiy-Yakovenko** [comment]
This one is a separate problem, and can be fixed on deadbeef side.

**Comment by orbea** [comment]
I found preexisting Gentoo issue and made a PR for Gentoo backporting this commit.

https://bugs.gentoo.org/870187
https://github.com/gentoo/gentoo/pull/27377
